File: rust/src/bin/2015-01.rs
use my_project::get_input;

fn main() {
    let content = get_input(2015, 1).expect("Can't get the content");

    let mut floor = 0;
    let mut res_b = None;

    for (id, inst) in content.bytes().enumerate() {
        match inst {
            b'(' => floor += 1,
            b')' => floor -= 1,
            _ => {}
        }
        if res_b.is_none() && floor < 0 {
            res_b = Some(id + 1);
        }
    }

    println!("Part A: {}", floor);
    println!("Part B: {}", res_b.unwrap_or(usize::MAX));
}

File: python/2015-01.py
from inputGetter import get_input

content = get_input(2015, 1)

floor = 0
check = True
res_b = -1

for id in range(len(content)):
    floor += 1 if content[id] == '(' else -1
    if check:
        if floor < 0:
            check = False
            res_b = id

print(f'Part A: {floor}')
print(f'Part B: {res_b}')